sys/drv/pit.c

```diff
@@ -121,7 +122,8 @@ static int pit_attach(device_t *dev) {
   assert(pit->regs != NULL);
 
   pit->lock = SPIN_INITIALIZER(0);
-  pit->intr_handler = INTR_HANDLER_INIT(pit_intr, NULL, pit, "i8254 timer", 0);
+  /* intr_handler name: "i8254 timer" */
+  pit->irq_res = bus_alloc_irq(dev, 0, 0 /* magic */, RF_ACTIVE);
```

Comment:
...

Reply:
I'm counting on you to eliminate the need for the magic ;-)